engine/tabsqlitedb.py

```diff
@@ -465,8 +463,14 @@ def sync_usrdb (self):
         self.db.execute('PRAGMA wal_checkpoint;')
 
     def reset_phrases_cache (self):
-        if self._phrases_cache != None:
-            self._phrases_cache = {}
+        self._phrases_cache = {}
+
+    def invalidate_phrases_cache (self, tabkeys=u''):
+        for i in range(1, self._mlen + 1):
+            try:
+                self._phrases_cache.pop(tabkeys[0:i])
+            except:
+                pass
```

Comment:
Maybe 
    if self._phrase_cache.get(tabkeys[0:i]:
        self._phrases_cache.pop(tabkeys[0:i])
is nicer than the "try" and "except".